File: LeetCode/0007.reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        long reverseNum=0;
        while(x!=0){
            reverseNum=reverseNum*10+x%10;
            x/=10;
        }
        if(reverseNum>Integer.MAX_VALUE || reverseNum<Integer.MIN_VALUE){
            return 0;
        }
        return (int)reverseNum;
    }
}

File: LeetCode/0007.reverse-integer/reverse-integer.kt
class Solution {
    fun reverse(x: Int): Int {
        var res = 0L
        var value : Int = x
        while (value != 0){
            res = res * 10 + value % 10
            value /= 10
        }
        if (res > Int.MAX_VALUE || res < Int.MIN_VALUE){
            return 0
        }
        return res.toInt()
    }
}